File: src/main/java/architecture/goldenboughs_lib/mixed/client/IModelBuilder.java
package architecture.goldenboughs_lib.mixed.client;

import architecture.goldenboughs_lib.api.NoMixinException;
import net.neoforged.neoforge.client.model.generators.ModelBuilder;

import java.util.Map;

public interface IModelBuilder<T extends ModelBuilder<T>> {
	static <T extends ModelBuilder<T>> IModelBuilder<T> of(ModelBuilder<T> obj) {
		return (IModelBuilder<T>) obj;
	}

	default Map<String, String> goldenboughs_lib$getTexture() {
		throw new NoMixinException();
	}
}

File: src/main/kotlin/architecture/goldenboughs_lib/mixed/client/IModelBuilder.kt
package architecture.goldenboughs_lib.mixed.client

import architecture.goldenboughs_lib.api.NoMixinException
import net.neoforged.neoforge.client.model.generators.ModelBuilder

interface IModelBuilder<T : ModelBuilder<T>> {
	fun `goldenboughs_lib$getTexture`(): MutableMap<String, String> {
		throw NoMixinException()
	}

	companion object {
		@JvmStatic
		fun <T : ModelBuilder<T>> of(obj: ModelBuilder<T>): IModelBuilder<T> {
			return obj as IModelBuilder<T>
		}
	}
}
